src/main/java/SearchResultPage.java

```diff
@@ -0,0 +1,19 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+
+public class SearchResultPage {
+
+    WebDriver driver;
+    By firstSwitchSelector = By.cssSelector("[data-component-type='s-search-result']");
+    public SearchResultPage(WebDriver driver){
+        this.driver = driver;
+    }
+
+    public ProductPage OpenProductPage(){
```

Comment:
can you modify this function to make it generic? 
`public ProductPage OpenProductPage(int index)`